### ISSUE TITLE
Consider rendering man_made=silo and man_made=storage_tank as buildings

### ISSUE THREAD

**Issue by jeisenbe** [gravitystorm/openstreetmap-carto#3967]
### Expected behavior
- Silos and storage tanks are types of built structures, similar to bridges and buildings which are currently rendered with a brown fill and outline
- Sometimes the wiki has recommended not double-tagging with building=* which is debatable but only about half have a building tag. 
- They (and man_made=gasometer) could also be rendered with the standard building polygon fill and outline
- A good comparison is man_made=bridge which is not a building (it does not have a roof or walls) but is clearly a man-made structure and is rendered in similar way 

### Actual behavior
- `silo` and `storage_tank` are rendered with an icon only
- However, the icon is not very intuitive (see #3927 and #3945 for discussion), since it needs to cover both silos and storage tanks.

**Comment by Adamant36** [comment]
I thought there was a conversation about this awhile back on like the mailing list or something around the same time icon was added. Unfortunately I can't find it, but if I remember correctly the consensus seemed to be that storage tanks didn't count as buildings because of smaller ones like propane tanks and neither do silos because some of them are skeleton's or something like that (I could be thinking of something else on that one though). Anyone with a better memory then me remember where that discussion happened or if it even took place?

**Comment by bhousel** [comment]
The thing I wrote on  openstreetmap/iD#4265:  (2018)

OSM evolves over time. These days I consider `building` as "any structure with a height that should render on a 3D map". This convention has been established for years on the OSM wiki and in software, so lots of things in OSM are now mapped as "buildings" that a layperson would not consider as buildings (like suspension bridge towers, silos, gasometers, storage tanks, gazebos).

This is fine. Similarly, nobody would consider a dirt footpath as a "highway", but anyway that is the tag and people accept it. To be pedantic, `building` should really be renamed `structure`, but we don't have a way to rename tags in OSM at that scale.

Look at it this way too: if you wanted to detail map a storage tank in 3D, you'd probably use `building:part`, not `storage_tank:part`.

(tl;dr: yes they should be rendered as buildings)

**Comment by jeisenbe** [comment]
> tl;dr: yes they should be rendered as buildings

Thank you for the clarification

>  if I remember correctly the consensus seemed to be that storage tanks didn't count as buildings

Right, and that means that they don't require a `building=*` tag. But we can certainly render them the same as buildings or in a very similar fashion, since they are structures. 

Alternatively we could use the bridge fill color `#B8B8B8` since they are structures that are not exactly buildings.

But I see them as more similar to buildings than bridges: unlike bridges which lack walls and roofs, storage tanks and silos are fully enclosed spaces, more like garages or storage sheds or barns (types of buildings) than like bridges or towers.

**Comment by Adamant36** [comment]
> we can certainly render them the same as buildings or in a very similar fashion, since they are structures.

I'd say in a similar fashion. Maybe not exactly the same. There are round "buildings" out there and be good to be able to tell them apart. There was that whole thing with rendering "minor" and "major" buildings slightly differently a while back and I thought that got axed because they to similar in rendering or that kind of thing just wasn't "in." So, rendering storage tanks as slightly off buildings might be kind of the same thing and run afoul for the same reasons. The bridge color might be good/different enough to pass, but then again maybe not.

**Comment by jeisenbe** [comment]
Also see #2620 - we could rendering man_made=reservoir_covered in the same way

**Comment by jeisenbe** [comment]
Also see #3530 - [man_made=gasometer](https://wiki.openstreetmap.org/wiki/Tag:man_made%3Dgasometer) could be rendered in the building style. Half are currently tagged with building=* - see https://taginfo.openstreetmap.org/tags/?key=man_made&value=gasometer#combinations

**Comment by nitinatsangsit** [comment]
I've seen no reason not to render a polygon of any `man_made`=* things without a `building` tag.
This has forced some mappers to add a `building` tag to every `man_made` object.

**Comment by Adamant36** [comment]
> his has forced some mappers to add a building tag to every man_made object.

Hardly. For example no one is adding the building to objects tagged as man_made=cross. Their just adding it to things that are either clearly buildings or could be considered ones.

**Comment by nitinatsangsit** [comment]
> For example no one is adding the building to objects tagged as man_made=cross.

From taginfo, `man_made=cross` has been tagged as a node for 10017 out of 10039. This means almost all of them would be rendered as an icon only, which makes sense. If some `cross` is quite large and could be mapped as an outline, rendering of their outline might be OK.
I'm referring to some large `man_made` things, like `communications_tower`, `lighthouse`, `observatory`, `tower`, etc. Their polygon of a structure outline should be rendered in some way, whether the same way as a building or not.

**Comment by Adamant36** [comment]
> 'm referring to some large man_made things, like communications_tower, lighthouse, observatory, tower, etc. Their polygon of a structure outline should be rendered in some way, whether the same way as a building or not.

It's not wrong to tag a tower as a building though. That's pretty much what they are. So I don't get what the issue is with tagging them that way. Same goes with lighthouses and observatories. The only one that might be considered a building out of those three is a communications tower and that's really up to personal preference as to if they are or aren't buildings. I could see arguments for either position making sense depending on the communications tower.
